### DISCUSSION TITLE
Ultimate mode broken on Zephyrus Duo 16 (2023)

### DISCUSSION THREAD

**Discussion by moldimolt** [seerge/g-helper#862] [Q&A] [answered]
**Describe the bug**
On my ASUS Zephyrus Duo 16 (2023), enabling "Ultimate Mode" causes the primary display to no longer work; G-Helper reports `Laptop Screen: Off`. The secondary display remains on. I had thought this was related to #825, but the issue persists in v0.100.

**Clear scenario to Reproduce**
1. Open G-Helper
2. Enable Ultimate mode
3. Reboot
4. Observe the primary display no longer works unless the Nvidia graphics driver is uninstalled, forcing it back onto the Radeon integrated GPU.

**Expected behavior**
Ultimate mode will route the primary display output to the dGPU.

**App Logs**
[log.zip](https://github.com/seerge/g-helper/files/12138221/log.zip)

**Screenshots or screencasts**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 11 22H2 (22621.1992)

 - Laptop model: ASUS Zephyrus Duo 16 (2023) GX650PY-XS97

**Asus software**
- Armoury Crate (or its services installed): No.
- MyASUS installed: No.
- Other Asus services running in background: G-Helper reports 0 ASUS services running, and I wiped my drive and reinstalled Windows prior to setting up this device to avoid ASUS bloatware. However, ASUS ScreenXpert is installed for managing the secondary display overlay.

Thank you.

**Comment by seerge** [comment]
@moldimolt hello, do I understand correctly that screen is black as well (not just not detected in app?)

Cause in a log i see that windows simply don't report it as a screen at all for Ultimate mode

```
2023-07-23 03:41:12: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DISPLAYPORT_EXTERNAL: \\.\DISPLAY5 (your second screen)
```

And in Standard it's there
```
2023-07-23 03:43:46: NE160QDM-NM4:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023-07-23 03:43:46: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DISPLAYPORT_EXTERNAL: \\.\DISPLAY2
```
---

There are other users with DUO and it seem to be ok for them, so looks like it's something isolated to your device and probably caused by NVdia drivers.

Command to turn Ultimate mode ON or OFF is very simple and same for all devices (run in powershell as admin). 

0 - to enable Ultimate (reboot after) , 1 - to disable Ultimate (reboot after)

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090016, 0)
```

-----

1. Does powershell command work same as app ?
2. Can you install Armury and see if it works there ?

**Comment by moldimolt** [comment, marked as answer]
After several hours of troubleshooting, I've managed to identify how to reproduce this issue:

1. As you expected, this is Nvidia driver related, installing the latest Nvidia drivers (at time of writing, `536.67-notebook-win10-win11-64bit-international-dch-whql.exe`) will create this issue, but installing the [OEM drivers from ASUS](https://rog.asus.com/us/laptops/rog-zephyrus/rog-zephyrus-duo-16-2023-series/helpdesk/) (`Graphic_DCH_ROG_NVIDIA_Z_V31.0.15.2901_34212.exe`) does *not* fix the issue, even if you completely remove the drivers from Device Manager before installing. 
2. Instead, this [seemingly-unrelated binary for ScreenPad/ScreenXpert from ASUS](https://dlcdnets.asus.com/pub/ASUS/nb/Image/CustomComponent/ASUSScreenPad2Optimizer/ASUSScreenPad2Optimizer.zip) fixes the issue. 

I suspected the ScreenPad2Optimizer binary may just be kicking the device out of Ultimate mode and thus appearing to "fix" the issue, but I tested to ensure that wasn't the case by removing the iGPU drivers (which only disabled the secondary display that's always routed to the iGPU), and switching between Standard/Ultimate mode several times and the fix was persistent. Additionally, I wanted to make sure this wasn't a case of needing both the OEM drivers and whatever that ScreenPad2Optimizer binary does, so I tested installing the drivers from Nvidia (causing the issue again) and running the ScreenPad2Optimizer, and it successfully fixed the issue. 

This issue can likely be closed as it is clearly not G-Helper related, but if you have any insight as to what's happening here I would be thankful, as I don't understand why this works to fix the problem.

**Reply by brittany-dot** [reply]
Hi, I have been having an issue somewhat similar with my ASUS Zephyrus Duo 16 (2023) GX650PY, and it has honestly been a problem for the last couple of years. I understand that this thread is older now, but I would really like to share my experience in hopes of finally finding a solution to my issue, possibly.

My system has the RTX 4090 Laptop GPU, and I have had repeated problems with the NVIDIA GPU failing to initialize properly. In Windows, I have seen Device Manager show Code 43 on the NVIDIA GPU. I have also had situations where GPU-Z shows 0 MB VRAM / missing CUDA or OpenCL support, and in Linux, nvidia-smi fails to communicate with the GPU.

The strange part is that the behavior seems connected somehow to GPU mode / MUX behavior. Standard/Hybrid mode has sometimes allowed the system to behave better temporarily, but Ultimate mode tends to fail or causes display/artifact issues. I have also seen tearing/artifacts after the ASUS logo and have had DRIVER_POWER_STATE_FAILURE crashes.

I have tried multiple NVIDIA driver versions, including clean installs. Some older drivers seem to reduce the artifacts, but the Code 43 / GPU initialization issue remains. I have also tried flashing the correct ASUS vBIOS for the machine, but the problem still persists.

After reading this thread, I am wondering if there could be some connection between Ultimate mode, the ScreenPad/ScreenXpert components, and how the dGPU path is initialized on the Zephyrus Duo 16. I noticed that in this thread, installing/running the ScreenPad-related ASUS component seemed to restore proper Ultimate mode behavior, which makes me wonder if this is not only a NVIDIA driver issue, but possibly a display-routing / ASUS service / MUX initialization issue.

I know this may not be directly related to G-Helper, but I wanted to ask here because this is one of the only discussions I have found that sounds close to what I am seeing. Do you happen to know which ASUS services, ScreenPad/ScreenXpert components, or OEM packages are actually required for the Duo 16 to switch correctly between Standard and Ultimate mode?

Any insight would be greatly appreciated. I am trying to determine whether my issue is truly failed GPU hardware or whether the GPU is failing to initialize because of a missing ASUS component, MUX/display routing issue, firmware issue, or power state problem.

**Comment by seerge** [comment]
Oke, I will move this to "discussions" instead then (so it remains in a knowledge base) :) 

Formally closing as completed.
